cluster-autoscaler/cloudprovider/digitalocean/digitalocean_node_group.go

```diff
@@ -144,7 +144,7 @@ func (n *NodeGroup) DecreaseTargetSize(delta int) error {
 	}
 
 	targetSize := n.nodePool.Count + delta
-	if targetSize <= n.MinSize() {
+	if targetSize < n.MinSize() {
```

Comment:
For a future PR, I wonder if the target size should be adjusted to the min size if it goes below instead of returning error.

Reply:
Like this @andrewsykim ?

```
if targetSize < n.MinSize() {
    targetSize = n.MinSize()
}
```

Reply:
Maybe a log message too

Reply:
Yeah, I'm not sure what the other providers do but that seems more correct then returning an error here.

Reply:
Talked to @waynr a bit. Maybe the error is relied upon somewhere else within the autoscaler and removing it may lead to undesired effects 🤔 

Not sure though. Maybe worth exploring.

Reply:
Whoops, meant to respond to this earlier. Returning an error when we can't decrease by the requested amount seems to me like the most intuitive behavior. That being said, this is my first contribution to autoscaler and I only have a passing familiarity with the principles of its operation so maybe it is okay for functions like this to only perform partial/"best effort" mutations on the cluster.

Reply:
> Maybe the error is relied upon somewhere else within the autoscaler and removing it may lead to undesired effects thinking

Yeah I don't have the wholistic picture of this but I think worth investigating.